ProcessList.c

```diff
@@ -392,14 +402,34 @@ void ProcessList_rebuildPanel(ProcessList* this) {
 
    const int processCount = Vector_size(this->displayList);
    int idx = 0;
+
+   // Mark Children and Parent for F4-Filter
+   Hashtable* filteredProcesses = Hashtable_new(processCount, false);
+   for (int i = 0; i < processCount; i++) {
+      Process* p = (Process*)Vector_get(this->processes, i);
+      pid_t ppid = Process_getParentPid(p);
+
+      if (incFilter && !(String_contains_i(Process_getCommand(p), incFilter, true)))
+         continue;
+
+      if (this->settings->showChildrenInFilter)
+         ProcessList_filterChildren(this, p->pid, filteredProcesses);
+
+      do {
+         Hashtable_put(filteredProcesses, p->pid, (void*) 1);
+         ppid = Process_getParentPid(p);
+         p = Hashtable_get(this->processTable, ppid);
+      } while (this->settings->showParentsInFilter && p && p->pid != p->ppid);
```

Comment:
As we know from #1017 (which I still need to rebase/reimplement), in some cases two processes may be parents of each other and thus this loop becomes an infinite one.

Reply:
got it, 
I will keep an eye on the progress of that issue